store-api-server/src/payment/service/payment.service.ts

```diff
@@ -403,6 +419,162 @@ WHERE nft_order.id = $1
     };
   }
 
+  async #createSimplexPaymentIntent(
+      userAddress: string,
+      fiatCurrency: string,
+      currencyUnitAmount: number,
+      usr: UserEntity
+  ): Promise<PaymentIntentInternal> {
+    if (
+        typeof SIMPLEX_API_URL === "undefined" ||
+        typeof SIMPLEX_API_KEY === "undefined" ||
+        typeof SIMPLEX_WALLET_ID === "undefined"
+    ) {
+      throw new HttpException(
+          "Simplex payment provider not supported by this API instance",
+          HttpStatus.NOT_IMPLEMENTED
+      );
+    }
+    if (!SIMPLEX_ALLOWED_FIAT.includes(fiatCurrency)) {
+      throw new HttpException(
+          `requested fiat (${fiatCurrency}) is not supported by Wert`,
+          HttpStatus.BAD_REQUEST
+      );
+    }
+    const decimals = SUPPORTED_CURRENCIES[fiatCurrency];
+    const amount = (Number(currencyUnitAmount) * Math.pow(10, -decimals)).toFixed(
+        decimals
+    )
+    let quoteId;
+    try {
```

Comment:
I put this actually I can handle exactly response from Simplex API integration. Because the error is ambiguous from simplex I want to handle this. Is it correct?

Reply:
Sorry maybe I wasn't clear. I didn't mean to ask to remove the try catch, rather wanted to indicate that each of those try catch blocks should be moved into a separate function. Not because try catches inherently belong in a function, but because in this case these blocks are doing something specific that really should belong in a dedicated function for clarity sake.

Reply:
yes I understood actually we applied SOLID principles for this methods. Single Responsibility comes first :) I extract methods for two operations